src/sample.jl

```diff
@@ -284,8 +284,23 @@ function mcmcsample(
     nchains::Integer;
     progress = PROGRESS[],
     progressname = "Sampling ($(min(nchains, Threads.nthreads())) threads)",
+    rngs::Union{Nothing, AbstractVector{<:Random.AbstractRNG}} = nothing,
     kwargs...
 )
+    interval = 1:min(nchains, Threads.nthreads())
+    if rngs === nothing
+        # Copy the random number generator, model, and sample for each thread
+        # NOTE: As of May 17, 2020, this relies on Julia's thread scheduling functionality
+        #       that distributes a for loop into equal-sized blocks and allocates them
+        #       to each thread. If this changes, we may need to rethink things here.
+        subrngs = [deepcopy(rng) for _ in interval]
+        # Create a seed for each chain using the provided random number generator.
+        seeds = rand(rng, UInt, nchains)
+    else
+        subrngs = rngs
+        @assert length(rngs) == nchains
```

Comment:
In principle it would be sufficient to only provide `min(nchains, Threads.nthreads())` RNGs - why do we require `nchains` RNGs?

I think also we should use `error` or e.g. `ArgumentError` with a clear error message instead of `@assert` since we really want to check user input here and not some internal functionality:
```suggestion
        length(rngs) == nchains || error("the number of chains and random number generators has to be equal")
```
Also,